services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/AbstractPushMonitor.java

```diff
@@ -1120,15 +1123,45 @@ private void updateStoreVersionStatus(String storeName, int versionNumber, Versi
                     storeName,
                     versionNumber));
           }
-          if (version.isVersionSwapDeferred()) {
+
+          /**
+           * Switch to the new version if:
+           * 1.deferred version swap is not enabled and it is not a target region push w/ deferred version swap
+           * 2.target region push w/ deferred swap is enabled and the current region matches the target region
+           *
+           * Do not switch to the new version now if:
+           * 1.deferred version swap is enabled (it will be manually swapped at a later date)
+           * 2.target region push is enabled and the current region does NOT match the target region (it will be swapped
+           *  after targetSwapRegionWaitTime passes)
+           */
+          Set<String> targetRegions = RegionUtils.parseRegionsFilterList(version.getTargetSwapRegion());
+          boolean isTargetRegionPushWithDeferredSwap =
+              version.isVersionSwapDeferred() && targetRegions.contains(regionName);
+          boolean isNormalPush = !version.isVersionSwapDeferred();
+          boolean isDeferredSwap = version.isVersionSwapDeferred() && targetRegions.isEmpty();
+          if (isTargetRegionPushWithDeferredSwap || isNormalPush) {
+            LOGGER.debug(
+                "Swapping to version {} for store {} in region {}. isTargetRegionPushWithDeferredSwap is {} and"
+                    + "isNormalPush is {}",
+                versionNumber,
+                store.getName(),
+                regionName,
+                isTargetRegionPushWithDeferredSwap,
+                isNormalPush);
+            int previousVersion = store.getCurrentVersion();
+            store.setCurrentVersion(versionNumber);
+            realTimeTopicSwitcher.transmitVersionSwapMessage(store, previousVersion, versionNumber);
+          } else if (isDeferredSwap) {
             LOGGER.info(
                 "Version swap is deferred for store {} on version {}. Skipping version swap.",
                 store.getName(),
                 versionNumber);
           } else {
-            int previousVersion = store.getCurrentVersion();
-            store.setCurrentVersion(versionNumber);
-            realTimeTopicSwitcher.transmitVersionSwapMessage(store, previousVersion, versionNumber);
+            LOGGER.info(
+                "Version swap is deferred for store {} on version {} in region {} due to target region push w/ deferred swap",
```

Comment:
may be log why its deferred, eg deferred as target region {} is different from the region {} running this push.